lib/fastifySession.js

```diff
@@ -229,6 +229,25 @@ function fastifySession (fastify, options, next) {
   function option (options, key, def) {
     return options[key] === undefined ? def : options[key]
   }
+
+  function verifyPath (path, cookiePath) {
+    if (path === cookiePath) {
+      return true
+    }
+    const pathLength = path.length
+    const cookiePathLength = cookiePath.length
+
+    if (pathLength <= cookiePathLength) {
+      return false
+    } else if (path.startsWith(cookiePath)) {
+      if (path[cookiePathLength] === '/') {
```

Comment:
do we need this `if`? seems to behave the same as the one 2 lines below

Reply:
> do we need this `if`? seems to behave the same as the one 2 lines below

I have through a while, but it actually does require.

The first assert `/docs` with `/docs/(.+)`, the second assert `/docs/` with `/docs/(.+)`.

Reply:
E.g. 

```
path = /check/
cookiePath = /check
```

`path[cookiePathLength]` would be '/' 

So we know that the path is terminated with a '/'

The below one covers another case

```
path = /check/index
cookiePath = /check/
```

So we know from line 242 that path startsWith cookiePath. So we know that path starts with `/check/`. But the line 243 would return false, because `path[cookiePathLength]` would be `i` ( /check/**i**ndex). So with line 245 we check if cookiePath is terminated with a '/' and if so we can say that the path is valid.

Reply:
cool, thanks for thorough explanation 👍